ingest/src/test/java/com/microsoft/azure/kusto/ingest/StreamingIngestClientTest.java

```diff
@@ -0,0 +1,450 @@
+
+        // Create Compressed CSV stream and ingest
+        ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+        GZIPOutputStream gzipOutputStream = new GZIPOutputStream(byteArrayOutputStream);
+        byte[] inputArray = Charset.forName("UTF-8").encode("Name, Age, Weight, Height").array();
+        gzipOutputStream.write(inputArray, 0, inputArray.length);
+        gzipOutputStream.flush();
+        gzipOutputStream.close();
+        inputStream = new ByteArrayInputStream(byteArrayOutputStream.toByteArray());
+        streamSourceInfo.setStream(inputStream);
+        streamSourceInfo.setIsCompressed(true);
+        status = streamingIngestClient.ingestFromStream(streamSourceInfo, ingestionProperties).getIngestionStatusCollection().get(0).status;
+        assert status == OperationStatus.Succeeded : "Ingestion failed with status: " + status.toString();
+
+        // Create JSON stream from String and ingest
+        data = "{\"Name\": \"name\", \"Age\": \"age\", \"Weight\": \"weight\", \"Height\": \"height\"}";
+        inputStream = new ByteArrayInputStream(Charset.forName("UTF-8").encode(data).array());
+        streamSourceInfo = new StreamSourceInfo(inputStream);
+        ingestionProperties.setDataFormat(IngestionProperties.DATA_FORMAT.json);
+        ingestionProperties.setIngestionMapping("JsonMapping", IngestionMapping.IngestionMappingKind.json);
+        status = streamingIngestClient.ingestFromStream(streamSourceInfo, ingestionProperties).getIngestionStatusCollection().get(0).status;
+        assert status == OperationStatus.Succeeded : "Ingestion failed with status: " + status.toString();
+
+        // Create Compressed JSON stream and ingest
+        byteArrayOutputStream = new ByteArrayOutputStream();
+        gzipOutputStream = new GZIPOutputStream(byteArrayOutputStream);
+        inputArray = Charset.forName("UTF-8").encode("{\"Name\": \"name\", \"Age\": \"age\", \"Weight\": \"weight\", \"Height\": \"height\"}").array();
+        gzipOutputStream.write(inputArray, 0, inputArray.length);
+        gzipOutputStream.flush();
+        gzipOutputStream.close();
+        inputStream = new ByteArrayInputStream(byteArrayOutputStream.toByteArray());
+        streamSourceInfo.setStream(inputStream);
+        streamSourceInfo.setIsCompressed(true);
+        status = streamingIngestClient.ingestFromStream(streamSourceInfo, ingestionProperties).getIngestionStatusCollection().get(0).status;
+        assert status == OperationStatus.Succeeded : "Ingestion failed with status: " + status.toString();
+    }
+
+    @Test
+    void IngestFromStream_StreamSourceInfo_IllegalArgumentException() {
+        assertThrows(IllegalArgumentException.class,
+                () -> streamingIngestClient.ingestFromStream(null, ingestionProperties),
+                "Expected IllegalArgumentException to be thrown, but it didn't");
+    }
+
+    @Test
+    void IngestFromStream_IngestionProperties_IllegalArgumentException() {
```

Comment:
There're a bunch of tests here (and in methods below) with generic messages, how will you know what's wrong when something will fail?
One option would be to separate those into different tests